### PR TITLE
Add Load Balancers Support

### PR DESCRIPTION
**Pull request by iheanyi**
This pull request adds in support for load balancers inside of the `godotto` package. :)

### CODE REVIEW

File: cmd/dorepl/main.go [hunk 1]
```diff
@@ -181,5 +181,9 @@ func enumerateLeftover(spy func(...spycloud.Spy)) {
 			once.Do(print)
 			log.Printf("- Tag: %q", v.Name)
 		}),
+		spycloud.LoadBalancers(func(v *godo.LoadBalancer) {
+			once.Do(print)
+			log.Printf("- Load Balancer: Q5", v.Name)
```
Comment (viola) [inline cmd/dorepl/main.go]: `Q5` => `%q`

File: internal/godojs/godojs.go [hunk 2]
```diff
@@ -161,6 +161,142 @@ func ArgTagUntagResourcesRequest(vm *otto.Otto, v otto.Value) *godo.UntagResourc
 	return req
 }
 
+func ArgLoadBalancerCreateRequest(vm *otto.Otto, v otto.Value) *godo.LoadBalancerRequest {
+	req := &godo.LoadBalancerRequest{
+		Name:                ottoutil.String(vm, ottoutil.GetObject(vm, v, "name", true)),
+		Algorithm:           ottoutil.String(vm, ottoutil.GetObject(vm, v, "algorithm", false)),
+		Region:              ArgRegionSlug(vm, ottoutil.GetObject(vm, v, "region", false)),
```
Comment (viola) [inline internal/godojs/godojs.go]: Region slug should be mandatory.

File: pkg/extra/do/cloud/loadbalancers/client.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+import (
+	"context"
+
+	"github.com/aybabtme/godotto/internal/godoutil"
+	"github.com/digitalocean/godo"
+)
+
+type Client interface {
+	Create(ctx context.Context, name, region string, forwardingRules []godo.ForwardingRule, opts ...CreateOpt) (LoadBalancer, error)
```
Comment (viola) [inline pkg/extra/do/cloud/loadbalancers/client.go]: Missing `Update`

**Comment by viola** [comment]
@iheanyi this is looking great! left few minor comments but feel free to reach out for more 👀  as I'm not that familiar with repo!

**Comment by iheanyi** [comment]
Thanks @viola :)

File: internal/godojs/godojs.go [hunk 4]
```diff
@@ -161,6 +161,156 @@ func ArgTagUntagResourcesRequest(vm *otto.Otto, v otto.Value) *godo.UntagResourc
...
+	return req
+}
+
+func ArgLoadBalancerUpdate(vm *otto.Otto, v otto.Value) *godo.LoadBalancerRequest {
+	return &godo.LoadBalancerRequest{
+		Name:                ottoutil.String(vm, ottoutil.GetObject(vm, v, "name", true)),
+		Algorithm:           ottoutil.String(vm, ottoutil.GetObject(vm, v, "algorithm", false)),
+		Region:              ArgRegionSlug(vm, ottoutil.GetObject(vm, v, "region", false)),
+		DropletIDs:          ArgDropletIDs(vm, ottoutil.GetObject(vm, v, "droplet_ids", false)),
+		HealthCheck:         ArgHealthCheck(vm, ottoutil.GetObject(vm, v, "health_check", false)),
```
Comment (viola) [inline internal/godojs/godojs.go]: for the update health check is mandatory

File: internal/godojs/godojs.go [hunk 5]
```diff
@@ -161,6 +161,156 @@ func ArgTagUntagResourcesRequest(vm *otto.Otto, v otto.Value) *godo.UntagResourc
...
+	}
+
+	return req
+}
+
+func ArgLoadBalancerUpdate(vm *otto.Otto, v otto.Value) *godo.LoadBalancerRequest {
+	return &godo.LoadBalancerRequest{
+		Name:                ottoutil.String(vm, ottoutil.GetObject(vm, v, "name", true)),
+		Algorithm:           ottoutil.String(vm, ottoutil.GetObject(vm, v, "algorithm", false)),
+		Region:              ArgRegionSlug(vm, ottoutil.GetObject(vm, v, "region", false)),
```
Comment (viola) [inline internal/godojs/godojs.go]: region slug should be mandatory

**Comment by iheanyi** [comment]
Thanks @viola!

**Comment by viola** [comment]
@iheanyi LGMT

File: internal/godojs/godojs.go [hunk 6]
```diff
@@ -738,6 +888,85 @@ func TagToVM(vm *otto.Otto, g *godo.Tag) otto.Value {
...
+		"redirect_http_to_https": g.RedirectHttpToHttps,
+	})
+}
+
+func ForwardingRulesToVM(vm *otto.Otto, g []godo.ForwardingRule) otto.Value {
+	if g == nil {
+		return otto.NullValue()
+	}
+
+	var rules = make([]map[string]interface{}, 0)
```
Comment (aybabtme) [inline internal/godojs/godojs.go]: `rules := make([]map[string]interface{}, 0)` or `var rules []map[string]interface{}` but not both :P

File: pkg/loadbalancers/loadbalancers.go [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+	if err != nil {
+		ottoutil.Throw(vm, err.Error())
+	}
+	return q
+}
+
+func (svc *loadBalancersSvc) list(all otto.FunctionCall) otto.Value {
+	vm := all.Otto
+
+	var lbs = make([]otto.Value, 0)
```
Comment (aybabtme) [inline pkg/loadbalancers/loadbalancers.go]:
```go
lbs := make([]otto.Value, 0) // if you _must_ have `lbs` never be `nil`
var lbs []otto.Value // the rest of the time
```
but don't mix both

**Comment by aybabtme** [comment]
:shipit: whenever

**Pull request merged**